src/main/java/firebase/sdk/dashboard/dao/SDKDaoDatastore.java

```diff
@@ -0,0 +1,343 @@
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.QueryResultIterable;
+
+/**
+ * An interface for retrieving and manipulating data about SDKs.
+ */
+public class SDKDaoDatastore implements SDKDao {
+
+  private static final DatastoreService DATASTORE = DatastoreServiceFactory.getDatastoreService();
+
+  public SDK getSDK(Platform givenPlatform, String givenLibraryName) {
+    Key sdkKey = KeyFactory.createKey("SDK", givenPlatform.toString() + "_" + givenLibraryName);
+
+    FilterPredicate keyPropertyFilter = makePropertyFilter("__key__", sdkKey);
+
+    Query query = new Query("SDK")
+      .setFilter(keyPropertyFilter);
+
+    PreparedQuery preparedQuery = DATASTORE.prepare(query);
+    Entity entity = preparedQuery.asSingleEntity();
+
+    if (entity == null) {
+      return null;
+    }
+
+    Platform platform = Platform.fromInteger(((Number) entity.getProperty("platform")).intValue());
+    String libraryName = (String) entity.getProperty("libraryName");
+    String libraryGroup = (String) entity.getProperty("libraryGroup");
+    String externalName = (String) entity.getProperty("externalName");
+    String fireEscapeName = (String) entity.getProperty("fireEscapeName");
+    String owner = (String) entity.getProperty("owner");
+    List<VersionMetadata> versionHistory = getVersionHistoryFromProperty(
+        (List<EmbeddedEntity>) entity.getProperty("versionHistory"));
+
+    SDK sdk = SDK.newBuilder()
+      .platform(platform)
+      .libraryName(libraryName)
+      .libraryGroup(libraryGroup)
+      .externalName(externalName)
+      .fireEscapeName(fireEscapeName)
+      .owner(owner)
+      .versionHistory(versionHistory)
+      .build();
+
+    return sdk;
+  }
+
+  public List<String> getSDKs(Platform givenPlatform) {
+    FilterPredicate platformPropertyFilter = makePropertyFilter("platform", givenPlatform.getValue());
+
+    Query query = new Query("SDK")
+      .setFilter(platformPropertyFilter);
+
+    PreparedQuery preparedQuery = DATASTORE.prepare(query);
+    FetchOptions fetchOptions = FetchOptions.Builder.withDefaults();
+    QueryResultIterable<Entity> results = preparedQuery.asQueryResultIterable();
+
+    if (results == null) {
+      return Arrays.asList();
+    }
+
+    ArrayList<String> sdksForPlatform = new ArrayList<>();
+    for (Entity entity: results) {
```

Comment:
Blank space is required on both sides of the colon:
https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace